rero_ils/modules/items/serializers/collector_ext.py

```diff
@@ -0,0 +1,426 @@
+            renewal_agg = A(
+                'filter',
+                term={'loan.trigger': ItemCirculationAction.EXTEND},
+                aggs=dict(item_pid=A(
+                    'terms', size=chunk_size, field='loan.item.pid')))
+            query.aggs.bucket('renewal', renewal_agg)
+            # adds last transaction aggregation for the fours triggers below.
+            triggers = [
+                ItemCirculationAction.CHECKOUT,
+                ItemCirculationAction.CHECKIN,
+                ItemCirculationAction.EXTEND
+            ]
+            loans_agg = A(
+                'filter', terms={'loan.trigger': triggers}, aggs=dict(
+                    item_pid=A(
+                        'terms',
+                        field='loan.item.pid',
+                        size=chunk_size,
+                        aggs=dict(last_op=A('max', field='date')))))
+            query.aggs.bucket('loans', loans_agg)
+            # query execution
+            result = query[0:0].execute()
+            # dump output into a dict
+            # checkouts data
+            items_stats = {
+                term.key: {
+                    'checkout_count': term.doc_count,
+                    'last_checkout': ciso8601.parse_datetime(
+                        term.last_op.value_as_string).date()}
+                for term in result.aggregations.checkout.item_pid
+            }
+            # renewal data
+            for term in result.aggregations.renewal.item_pid:
+                items_stats.setdefault(term.key, {})
+                items_stats[term.key]['renewal_count'] = term.doc_count
+            # last_transaction data
+            for term in result.aggregations.loans.item_pid:
+                items_stats.setdefault(term.key, {})
+                items_stats[term.key]['last_transaction'] = \
+                    ciso8601.parse_datetime(
+                        term.last_op.value_as_string).date()
+            return items_stats
+
+        chunk_pids = []
+        for pid in item_pids:
+            if len(chunk_pids) % chunk_size == 0:
+                stats = _get_loans_by_item_pids(chunk_pids)
+                for pid in chunk_pids:
+                    yield stats.get(pid, {})
+                chunk_pids = []
+            chunk_pids.append(pid)
+        stats = _get_loans_by_item_pids(chunk_pids)
+        for pid in chunk_pids:
+            yield stats.get(pid, {})
+
+    @staticmethod
+    def append_loan_data(hit, csv_data, items_stats):
+        """Append item loan.
+
+        :param hit: item ES record.
+        :param csv_data: dictionary of data.
+        :param loans: loans data.
+        """
+        stat = next(items_stats)
+        csv_data['item_checkouts_count'] = stat.get('checkout_count', 0)
+        csv_data['item_renewals_count'] = stat.get('renewal_count', 0)
+        csv_data['last_transaction_date'] = stat.get('last_transaction')
+        csv_data['last_checkout_date'] = stat.get('last_checkout')
```

Comment:
It appears that only `get_documents_by_item_pids()` is overloaded.
You could remove all these lines of code if you inherit of Collector class